gsy_framework/validators/profile_validator.py

```diff
@@ -50,3 +54,12 @@ def _validate_slot_length(self):
                 f"{self.slot_length.total_seconds() / 60} minutes "
                 f"(previous time_slot: {first_time})")
             first_time = time_slot
+
+    def _get_and_validate_time_diffs(self) -> timedelta:
+        first_time = self._profile_start_time
+        time_diffs = set()
+        for time_slot in list(self.profile.keys())[1:]:
+            time_diffs.add((time_slot-first_time).total_seconds())
+            first_time = time_slot
```

Comment:
Equally performant, but a set comprehension is more concise IMO:

```suggestion
    timestamps = list(self.profile.keys())
    time_diffs = {end - start for start, end in zip(timestamps[:, -1], timestamps[1:])}
```
The map function would equally do the trick, however I think that this is less memory-intensive.